Core/automation/lib/python/core/metadata.py

```diff
@@ -0,0 +1,193 @@
+'''
+Wrapper class for using Item Metadata
+'''
+
+from collections import MutableMapping
+from org.eclipse.smarthome.core.items import Metadata, MetadataKey
+from core.log import logging, LOG_PREFIX
+log = logging.getLogger(LOG_PREFIX + ".metadata")
+from core import osgi
+MetadataRegistry = osgi.get_service("org.eclipse.smarthome.core.items.MetadataRegistry")
+
+__all__ = [ "namespace_exists", "item_metadata", "metadata_namespace" ]
+
+
+def namespace_exists(item_name, namespace):
+    '''Returns True if metadata namespace exists'''
+    return False if MetadataRegistry.get(MetadataKey(namespace, item_name)) is None else True
+
+def _resolveType(value):
+    '''Attempts to resolve the type of the metadata value.
+    It will return the value as the python type if possible,
+    otherwise will return value as string'''
+    if str(value).lower().strip() == "true":
+        return True
+    elif str(value).lower().strip() == "false":
+        return False
+    else:
+        # attempt to cast to int
+        try: return int(str(value))
+        except ValueError: pass
+        # attempt to cast to float
+        try: return float(str(value))
+        except ValueError: pass
+        # not a number
+        return str(value)
+
+class item_metadata(MutableMapping):
+    def __init__(self, item_name):
+        '''This is a manager class that can be used to manage multiple
+        metadata namespaces for a single item from one place.
+        - This class behaves like a dict of "metadata_namespace" objects.
+        - Namespaces can be added by name with "add_namespace(name)"
+        (attempting to add a namespace that already exists will raise a KeyError)'''
+        self._item_name = item_name
+        self._namespaces = {}
+
+    def __iter__(self): return iter(self._namespaces)
+    def __len__(self): return len(self._namespaces)
+    def __setitem__(self, key, value): raise AttributeError("Direct assignment to keys not possible")
+    def __getitem__(self, key): return self._namespaces[key]
+    def __delitem__(self, key): self.delete_namespace(key)
+
+    def add_namespace(self, name):
+        '''Creates a new namespace instance and loads it from the registry
+        if it exists.'''
+        if name in self._namespaces: raise KeyError
+        else: self._namespaces[name] = metadata_namespace(self._item_name ,name)
+
+    def delete_namespace(self, name, remove=False):
+        '''Deletes the specified namespace from this object. It will not
+        remove the namespace from openHAB unless the "remove" flag is set'''
+        if name not in self._namespaces: raise KeyError
+        else:
+            if remove: self._namespaces[name].remove
+            self._namespaces.pop(name, None)
+
```

Comment:
extra empty line

Reply:
Like this for readability?
```python
        else:
            if remove: self._namespaces[name].remove
            
            self._namespaces.pop(name, None)
```

Reply:
No, no. :slightly_smiling_face: There are two empty lines there, so I was suggesting to remove one.

Reply:
Oh I see, I wondered about that while I was working on it last night. I usually do that between classes/functions and single line spaces between methods or subfuncs so it's easier to see the separation. I was inconsistent with that in the file though. So a preference for single blank lines throughout then?

Reply:
Yes, but I'm not sure if PEP8 says anything about it.

Reply:
PEP8 says what I do, but should we stick to one blank for consistency?

> Surround top-level function and class definitions with two blank lines.
>
> Method definitions inside a class are surrounded by a single blank line.
>
> Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (e.g. a set of dummy implementations).
>
> Use blank lines in functions, sparingly, to indicate logical sections.